okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -486,13 +487,23 @@ void onException(Throwable failureCause) {
    */
   private void onError(ErrorCode errorCode, String moreDetail) {
     frameWriter.goAway(0, errorCode, new byte[0]);
-    onGoAway(0, toGrpcStatus(errorCode).augmentDescription(moreDetail));
+    startGoAway(0, toGrpcStatus(errorCode).augmentDescription(moreDetail));
   }
 
-  private void onGoAway(int lastKnownStreamId, Status status) {
-    boolean notifyShutdown;
+  private void startGoAway(int lastKnownStreamId, Status status) {
+    synchronized (lock) {
+      if (goingAway) {
+        // Another go-away is inprocess, ignore this one.
```

Comment:
s/inprocess/in progress/
However, this description is still not accurate. `goingAway` will remain true even if the process is completed. I feel `isShutdown` would be a better name. Because `transportShutdown()` will be called afterwards, setting `isShutdown=true` here makes perfect sense. And if we name the variable as `isShutdown`, naming the method as something related to "shutdown" would make more sense.